typescript/src/schema/resource-database.ts

```diff
@@ -185,6 +185,32 @@ export class ResourceDatabase {
     return params;
   }
 
+  private getName(pattern: string): string {
```

Comment:
Should either be static, or just a helper function outside of the class (maybe it's better to move it to `util` and rename to `getResourceNameByPattern`?) because it does not use `this`.

Reply:
got it.